File: portal/src/main/resources/static/service/insuranceLookup/insuranceLookupService.js
function insuranceLookupService(MODULE_CONFIG,PATIENT_MODULE_CONFIG, $http) {
	this.addInsuranceLookup=function(insuranceLookup){
		return $http.post(PATIENT_MODULE_CONFIG.ADD_INSURANCELOOKUP(),insuranceLookup);
	};
	this.getAllInsuranceLookups=function(){
		return $http.get(PATIENT_MODULE_CONFIG.ALL_INSURANCE_LOOKUPS());
	};
	this.updateInsuranceLookup=function(insurancelookup){
		return $http.put(PATIENT_MODULE_CONFIG.UPDATE_INSURANCE_LOOKUP(),insurancelookup);
	};
	this.delteInsuranceLookup=function(insuranceLookupId){
		return $http.delete(PATIENT_MODULE_CONFIG.DELETE_INSURANCELOOKUP(insuranceLookupId));
	};
}

angular.module('HealthApplication').service('insuranceLookupService', insuranceLookupService);

File: patient/src/main/java/com/openspace/PatientManagement/service/InsuranceLookupService.java
package com.openspace.PatientManagement.service;

import java.util.List;

import com.openspace.Model.Lookups.InsuranceLookup;

public interface InsuranceLookupService {

	void add(InsuranceLookup insuranceLookup);

	List<InsuranceLookup> getAll();

	void deleteInsuranceLookup(Long insurancelookupId);

	void updateInsuranceLookup(InsuranceLookup insuranceLookup);

}
